### ISSUE TITLE
Add searchables section for titles and title collections

### ISSUE THREAD

**Issue by elv-serban** [eluv-io/elv-fabric-browser#91]
The site-level 'searchables' feature uses links to its title or title collections objects 'searchables' section.

We need to enhance the asset manager for title and title collections to create one or all 'searchables' links.

Ideally just present a section called 'searchables' and allow to user to select one or more of the supported categories.

Our current convention uses the following searchables categories:

1.  `asset_metadata`  - this is the most common and is a link to /public/asset_metadta
2. `offerings`  - this is the second most common and is a link to /offerings (not under /public)
3. `assets`   - link to /assets
4. `video_tags` - link to /video_tags

Example full searchables section:

```json
  "searchables": {
    "asset_metadata": {
      ".": {
        "auto_update": {
          "tag": "latest"
        },
      },
      "/": "./meta/public/asset_metadata"
    },
    "assets": {
      ".": {
        "auto_update": {
          "tag": "latest"
        },
      },
      "/": "./meta/assets"
    },
    "offerings": {
      ".": {
        "auto_update": {
          "tag": "latest"
        },
      },
      "/": "./meta/offerings"
    },
    "video_tags": {
      ".": {
        "auto_update": {
          "tag": "latest"
        },
      },
      "/": "./meta/video_tags"
    }
  }
```

## Configuration

We could use the content type JSON schema to determine the searchables categories to show:

For example for a Title:

- asset_metadata
   - link to: /public/asset_metadata
   - hint: Basic title metadata (title, synopsis, release date, cast, ...)
- assets
   - link to: /assets
   - hint: Only used if the title contains assets.
- video_tags:
   - link to: /video_tags
   - hint: If the content is tagged using the video understanding machine learning model, select this to allow video tags search
- offerings:
   - link to: /offerings
   - hint: Select if you want to search available offerings for the title.  Typically used when offerings have different permissions.

And for the Title Collection just use `asset_metadata` and `assets` (because offerings and video_tags don't pertain)

**Comment by elv-serban** [comment]
### Design Notes:

(discussion 2023-05-17)

#### Content Type Configuration:
   Info Fields
       Searchable paths:
           -    Label/Field   [  asset_metadata ]   Link  [ /public/asset_metadata]    Hint [ Basic title info ...]

#### Management App
    Info Tab
        Searchable paths:
               - [ ]  asset_metadata   
               - [ ]  offerings

**Comment by elv-victor-amigo** [comment]
I ingested 5 different titles and then added a brief synopsis to each of them (lorem ipsum) as well as merged the searchables metadata as stated in the above comment. Been using the `nobleflash` demo tenancy. 
https://share.1password.com/s#X0V4bpTH_dt9gNEQgTsbNzlklkMihYZcSswuS8fFs9c

Then, I created a Search index object (`iq__3Ryg5eotzgAEFL1tuxkwD8hBXbx3`) with content type `Title Collection` where I added the Searchables tab using the gear icon and saving the content type's spec. Finally, I added all five recently ingested titles inside the Searchables section with no issues. 
I did the same with the Site object (`iq__t5JiDQW2W2tyHUyLaAbwrtkENc6`) but also added all titles inside the Titles tab as well. 

After saving all my changes, I now see `site_map` containing all titles with `searchables` pointing to their latest versions hashes. Not sure how to proceed from here or how to link the site object with the newly created search index.

**Comment by elv-victor-amigo** [comment]
I've also changed the spec of both `Title` and `Title Collection` content types so that Searchable Links flag is checked and necessary links are added:

 - Title:
<img width="877" alt="Screen Shot 2023-05-26 at 3 58 33 PM" src="https://github.com/eluv-io/elv-fabric-browser/assets/101947523/860718f6-7ce6-4612-8146-710c8a9790e0">

 - Title Collection:
<img width="892" alt="Screen Shot 2023-05-26 at 3 59 24 PM" src="https://github.com/eluv-io/elv-fabric-browser/assets/101947523/67d665ab-2db6-4dc3-a09e-096849312180">

**Comment by elv-victor-amigo** [comment]
Following up, I created a Site (`iq__2Cyy77DqNbw8m1XfBLX2JeDx8iFZ `) and Search Index (`iq__3mUrGK5J3u6cD6KZAwx1ksdvHxyJ `) content types inside the nobleflash tenancy. I then added all searchables titles I had previously set up inside the Searchables section of the Site object.

![Screen Shot 2023-05-30 at 1 23 38 PM](https://github.com/eluv-io/elv-fabric-browser/assets/101947523/857be26c-ad58-46b6-b65f-d151ad8d0fb1)

Then, I added the site object inside of the Search Index object, provided a document prefix so that it points to the entire object and added the necessary paths and types so that the index could query `title` and `synopsis` (on each video asset, I added a brief lorem ipsum synposis).

![Screen Shot 2023-05-30 at 1 24 59 PM](https://github.com/eluv-io/elv-fabric-browser/assets/101947523/4c7e1914-e553-4d5b-8182-82db9bce79e6)

Lastly, I updated the indexer and tested by running a query searching for a specific title string and text part of synopsis of the searchable objects. Worked without errors.

![Screen Shot 2023-05-30 at 1 28 21 PM](https://github.com/eluv-io/elv-fabric-browser/assets/101947523/c64db62c-f5c9-43a6-a7ce-62c76e82b65a)
